### DISCUSSION TITLE
Phased exposures - iterations - custom build

### DISCUSSION THREAD

**Discussion by Hazoooka** [sn4k3/UVtools#932] [Tutorials]
Hi Tiago.

I asked you some time ago to make optional wiping the layer with iterations in Phased Exposures.
Since you do not want to make the iterations work as before I have to make my own build.
I want to compile my custom build that suits my needs.
I just need informations.
I do not know your code and have no idea how it all works.
I need your help with the code, the compilation will be done by my friend.
Which parts of the code and in what files do I need to make changes to remove the safety from the layers being wiped, and let iterations to wipe the layers in Phased Exposures?
Please let me know everything I need to know to successfully compile my custom build.

Kind regards,
Witold

**Comment by sn4k3** [comment]
To allow empty layers all it needs is to comment or remove this line: https://github.com/sn4k3/UVtools/blob/f56dc0407dccc459ea4f033cb3efc4815295ab88/UVtools.Core/Operations/OperationPhasedExposure.cs#L364

**Comment by Hazoooka** [comment]
Thank you.
WIll do this.

Kind regards,
Witold

**Comment by Hazoooka** [comment]
Hi Tiago.

The compilation didn’t work out.
I can’t seem to find the necessary add-ons like AvaliniaVS.vis, dotnet…
I remember they were available on your site.
Perhaps I don’t look the right way.
Could you please send the link?

Kind regards,
Witold.

**Reply by sn4k3** [reply]
AvaloniaVS not required to compile, it's used only to see layout under the Visual Studio. All you need is dotnet sdk.

Directions [here](https://github.com/sn4k3/UVtools?tab=readme-ov-file#develop-and-build-from-source) for who wants to full develop.
Directions [here](https://github.com/sn4k3/UVtools/tree/master/build) for who wants easy edits and compile

I assume you on mac, so after having dependencies set all you will need is to call `bash createRelease.sh -b osx-arm64`

**Comment by Hazoooka** [comment]
Thank you!

**Comment by Hazoooka** [comment]
And what in case of Intel?
Would it be createRelease.sh -b osx-X86?

**Reply by sn4k3** [reply]
Close, but: `createRelease.sh -b osx-x64`

**Comment by Hazoooka** [comment]
Thank you!
i’m still on Intel
